test/common/stats/stat_test_utility.h

```diff
@@ -72,6 +75,38 @@ class MemoryTest {
   const size_t memory_at_construction_;
 };
 
+// Helps tests do stat-name lookups based on strings.
```

Comment:
@mattklein123 PTAL at this approach in lieu of the MixedStatNames in #9774 . If you like this direction I can fill it out with better doc, commenting, unit tests etc, covering findGauge() and histograms (currently in the interface but unimplemented).

The current impl has a problem which is that the maps are recomputed on demand only if the number of counters etc has changed. If someone drops and re-adds a scope with the same number of counters, the cache will be stale, we won't detect it, and that could result in accessing deleted old counters. I can just fix this by keeping an atomic sequence number in prod for all implementations of Store(). It's a little annoying because it adds (miniscule) cost to prod every time a new stat is allocated or deallocated(bump a counter). I think that's the option I favor.

I could also fix this by keeping shared_ptrs instead of references in the map. But that could affect code behavior by keeping the old allocated stats around when they should've been dropped and re-added.

Another option (more invasive) would be to change everywhere in source/... that instantiates an IsolatedStatsStore to be able instantiate my wrapper instead.  I think that's mostly common/upstream/upstream_impl.h , but it looks annoying to plug in an IsolatedStoreFactory.

WDYT?

Reply:
OK yeah in general I like this approach a lot better. It's much less surprising and IMO gives us a very clear path to getting rid of the string_view lookup methods in the prod code, right?

> The current impl has a problem which is that the maps are recomputed on demand only if the number of counters etc has changed.

Is this a premature optimization? It's test code? Does it matter?

Reply:
Is it a premature optimization that we keep string-maps, as opposed to vectorizing & searching each time? Guilty :)

Lemme see how things work with just vectorizing/searching each time. Could be in the noise compared to other test overhead. Maybe fuzz tests, where we care about speed, can use an entirely different mechanism.

I had also thought of letting the caller control this by instantiating the class (maybe call it LooupHelper) inline in the code when doing the lookups. Then the caller has explicit obvious control and mostly won't have to think of it. I was just thinking of how to make the code changes as syntactically compatible as possible, ideally with sed-script, to do the 1500-call-site-refactor down the road.

And yes, plan would be to get rid of `Stats::Scope::counter(const std::string&)` et al.

Reply:
done; this simplified the PR. We can always create some mechanism to do this with a map if there's a perf problem in tests.